packages/editor-ui/src/utils/nodeTypesUtils.ts

```diff
@@ -581,6 +583,66 @@ export function processNodeForAssistant(node: INode, propsToRemove: string[]): I
 	return nodeForLLM;
 }
 
+export function getNodeInfoForAssistant(node: INode) {
+	if (!node) {
+		return {};
+	}
+	const ndvStore = useNDVStore();
+	// Get all referenced nodes and their schemas
+	const referencedNodeNames = getReferencedNodes(node);
+	const schemas = getNodesSchemas(referencedNodeNames);
+
+	// Get node credentials details for the ai assistant
+	const nodeType = useNodeTypesStore().getNodeType(node.type);
+	let authType = undefined;
+	if (nodeType) {
+		const authField = getMainAuthField(nodeType);
+		const credentialInUse = node.parameters[authField?.name ?? ''];
+		const availableAuthOptions = getNodeAuthOptions(nodeType);
+		authType = availableAuthOptions.find((option) => option.value === credentialInUse);
+	}
+	let nodeInputData: { inputNodeName?: string; inputData?: IDataObject } | undefined = undefined;
+	const ndvInput = ndvStore.ndvInputData;
+	if (isNodeReferencingInputData(node) && ndvInput?.length) {
+		const inputData = ndvStore.ndvInputData[0].json;
+		const inputNodeName = ndvStore.input.nodeName;
+		nodeInputData = {
```

Comment:
what about multiple input nodes?

Reply:
This will send the input the user is seeing in ndv. Let me know if you think we should be getting everything but I think this is enough since we are providing user's visual context to the assistant here.

Reply:
If that's enough for our usecase then all good